### ISSUE TITLE
feat: add support for `snacks.nvim`

### ISSUE THREAD

**Issue by mikededo** [catppuccin/nvim#806]
**Is your feature request related to a problem? Please describe.**

[`snacks.nvim`](https://github.com/folke/snacks.nvim) is a new QoL plugin, used by default in LazyVim over a collection of separate plugins. Currently, the highlights provided by this plugin are not supported.

**Describe the solution you'd like**

To implement the highlight tokens specified in [here](https://github.com/folke/snacks.nvim?tab=readme-ov-file#-highlight-groups)

**Comment by rkaminsk** [comment]
Hi, I would also like to see an integration here too. Catppuccin does not play nice with snacks.notify because the borders of messages are set in italic, which looks ugly. My current workaround (for lazy.vim) is to patch some highlight groups:
```lua
return {
    "catppuccin/nvim",
    name = "catppuccin",
    priority = 1000,
    opts = {
      custom_highlights = function(colors)
        return {
          DiagnosticInfo = { style = {} },
          DiagnosticError = { style = {} },
          DiagnosticWarn = { style = {} },
        }
      end,
    },
  }

```

**Comment by mikededo** [comment]
Hey @rkaminsk, could you check if what's been implemented in #807 would work as you expect?

**Comment by rkaminsk** [comment]
@mikededo, yes your PR works.

PS: the default indicator, which is also a `|`, set with bufferline.nvim is also set in italic. Since all my attempts to configure this properly failed, I simply set
```vim
highlight BufferlineIndicatorSelected gui=bold
```
at the end of my config. This is of course unrelated to snacks.nvim.
